lib/inspec/fetcher/url.rb

```diff
@@ -118,6 +118,19 @@ def to_s
       @target
     end
 
+    def update_from_opts(opts)
+      changed = false
+
+      old_val = @archive_shasum
+      new_val = opts[:sha256]
+      unless old_val == new_val
+        @archive_shasum = new_val
+        changed = true
+      end
+
+      changed
```

Comment:
I hate to make "do something ugly and illogical" (having the opts not match the ivar name) the superclass implementation. To me, that obliges subclasses to override, which is not what you generally want.

Reply:
This doesn't make sense to me. You added the same named method to the superclass that returns `false` and thus "obliges subclasses to override". Can you clarify?

Having identical code in two sibling classes and pushing them up to the abstract superclass IS a very normal refactoring. This pushback doesn't make sense to me.

Reply:
First, thanks - you're right, I forgot I did provide a default superclass implementation that does no update and returns `false`. So any any subclass would be obliged to override if they wanted any behavior that wasn't that; but at least is it is a **sensible** default.

I'll cave on this one. Any new fetcher would need to remember to override that with a sane updater, because it's not likely they'll happen to choose the same ivar name that local and url did.